### PR TITLE
Prevent modifications on widget.currentSettings.formItems - create a …

### PR DESCRIPTION
**Pull request by frwg**
…deep copy of the formItems when opening featureEdit Dialog and process these copied formItems, rather than the original. This prevents bugs like unexpected data from a former featuer in the dialog

### CODE REVIEW

**Comment by frwg** [comment]
Fixing the following bug:
User opens a featureEditDialogue that contains an image file. Every following opened featureEditDialogue displays the same image, whether the underlying feature has an image set or not.

File: Resources/public/mapbender.element.digitizer.js [hunk 1]
```diff
@@ -998,7 +998,9 @@
                 schema.elementsTranslated = true;
             }
 
-            DataUtil.eachItem(widget.currentSettings.formItems, function(item) {
+            var formItems = JSON.parse(JSON.stringify(widget.currentSettings.formItems)); // Deep clone
```
Comment (LazerTiberius) [inline Resources/public/mapbender.element.digitizer.js]: Please use jQuery extend ( https://api.jquery.com/jquery.extend/ ) . JSON.parse(JSON.stringify()) can cause information lose,  and is very slow.

File: Resources/public/mapbender.element.digitizer.js [hunk 2]
```diff
@@ -998,7 +998,9 @@
                 schema.elementsTranslated = true;
             }
 
-            DataUtil.eachItem(widget.currentSettings.formItems, function(item) {
+            var formItems = [ $.extend(true,{},widget.currentSettings.formItems)[0]]; // Deep clone
```
Comment (LazerTiberius) [inline Resources/public/mapbender.element.digitizer.js]: Why not just $.extend(true,[], widget.currentSettings.formItems) ?

Comment (frwg) [reply Resources/public/mapbender.element.digitizer.js]: Well, that's what I did in cfc7282, but the root of `formItems` is supposed to be in array shape  in order to be processed by DataUtil.eachItem, while `$.extend` delivers a mere object.

Comment (LazerTiberius) [reply Resources/public/mapbender.element.digitizer.js]: Mind the second parameter in my comment. In cfc7282 you use the object literal and not an array one.

Comment (frwg) [reply Resources/public/mapbender.element.digitizer.js]: Ah, well, why not then. It's definitely more readable...

Comment (frwg) [reply Resources/public/mapbender.element.digitizer.js]: Now...

**Pull request merged**